crates/llm-base/src/inference_session.rs

```diff
@@ -191,7 +192,9 @@ impl InferenceSession {
         )?;
         stats.feed_prompt_duration = start_at.elapsed().unwrap();
         stats.prompt_tokens = self.n_past;
-
+        if request.run_perplexity {
+            self.perplexity(model, parameters, request, output_request);
```

Comment:
Consider adding a `perplexity: Option<f32>` to the `stats` (since they're available here), and setting it if `run_perplexity` is specified. You can then use these stats for printing in the CLI.